python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -640,13 +640,15 @@ def _kill_all_running_actors(self):
 
     def _kill_running_actor(self, actor: ray.actor.ActorHandle):
         """Kill the provided actor and remove it from the pool."""
-        ray.kill(actor)
         del self._num_tasks_in_flight[actor]
+        del self._actor_locations[actor]
```

Comment:
Could we add a note that we intend to refactor this with the pause API at some point?